small changes to 'getting started' and formatting in 'contributing'

jrscott opened the pull request.

Commit:
small changes to 'getting started' and formatting in 'contributing'

jahn committed:
fix and a couple more name changes

jahn committed:
fix a couple more typos

edoddridge committed:
link to contributing guide

and suggest that users work on a separate fresh copy of the code if they want to contribute pull requests.

edoddridge commented on doc/getting_started/getting_started.rst:
```diff
@@ -83,29 +83,31 @@ For specific checkpoint release ``XXX``, instead type:
-Need to figure out level of detail here... suggest workflow etc.?
+Git offers many tools to help organize and track changes in your work.  For example, one might keep separate projects on different branches, and update the code separately (using ``git pull``) on these separate branches. You can even make changes to code in the MIT repo tree; when git then tries to update code from upstream (see :numref:`git_setup`), it will notify you about possible conflicts an ...
 
+(we should be more specific above, need to agree how so however)
```
I think this is a good level of detail - but I realise that I'm not the target audience. Apart from adding a link to a git guide (perhaps this one http://git.huit.harvard.edu/guide/ ?), I think it's good.

edoddridge commented on doc/getting_started/getting_started.rst:
```diff
@@ -384,7 +399,8 @@ architecture) using the GCC (g77) compiler is
 If you write an optfile for an unrepresented machine or compiler, you
 are strongly encouraged to submit the optfile to the MITgcm project for
-inclusion. Please send the file to MITgcm-support@mitgcm.org. (**how about posting to maintainers somewhere on github instead?**)
+inclusion. Please submit the file through the `GitHub issue tracker <https://github.com/altMITgcm/MITgcm/issues>`_
```
I thought we decided that a pull request was our preferred method, and the others were backups?

edoddridge committed:
move mpirun example

edoddridge committed:
examples of reading output in MATLAB and Python

jrscott committed:
Merge branch 'master' of https://github.com/altMITgcm/MITgcm into manual_edits

jrscott committed:
more edits to getting started and contributing chapters

jrscott committed:
Merge branch 'master' of https://github.com/altMITgcm/MITgcm into manual_edits

jklymak commented on doc/contributing.rst:
```diff
@@ -33,33 +33,39 @@ Quickstart Guide
 
+   % git checkout master
+   % git pull upstream master
+   % git push origin master
```
A couple things on the github integration:
  - There is a community document that explains workflows that is meant to be linked into sphinx documents:  https://github.com/matthew-brett/gitwash  You may want to consider just including that instead of having your own set of instructions.  Not that the instructions here are bad, but `gitwash` was made to stop every group from typing these out each time...
  - That document suggests deleting your master branch on your local fork. http://matplotlib.org/devel/gitwash/development_workflow.html#consider-deleting-your-master-branch   I've found that super useful because it stops me from trying to add pull requests from my local master.

edoddridge replied:
Thanks @jklymak - we'll have a look at these.

jrscott committed:
changed ordering/section labels as per discussion

edoddridge committed:
improve steps for reviewing pull requests

Describe two methods for obtaining the code: cloning PR branch, or adding a remote.
Specify which directories one should be in when executing the code.

christophernhill commented:
+1 to merge - Chris
